eos_data_distribution/ndn/chunks.py

```diff
@@ -219,6 +219,14 @@ def _on_data(self, o, interest, data):
             self._qualified_name = name.getPrefix(-1)
 
         seg = get_chunk_component(name).toSegment()
+
+        # Have we somehow already got this segment?
+        if self._segments[seg] == SegmentState.COMPLETE:
+            logger.debug('Ignoring data ‘%s’ as it’s already been received',
+                         name)
+            self._check_for_complete()
+            return
```

Comment:
the code is good, but note that this is potentially very wrong, we should **never** get 2 segments (from the same host, but in general too) or we're being messy with our requests, or the other part is being messy with it's nacks.